iceoryx_posh/include/iceoryx_posh/popo/trigger_state.hpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2020 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_POSH_POPO_TRIGGER_STATE_HPP
+#define IOX_POSH_POPO_TRIGGER_STATE_HPP
+
+#include "iceoryx_posh/internal/log/posh_logging.hpp"
+#include "iceoryx_utils/cxx/function_ref.hpp"
+#include "iceoryx_utils/error_handling/error_handling.hpp"
+
+#include <cstdint>
+#include <limits>
+
+namespace iox
+{
+namespace popo
+{
+/// @brief TriggerState holds the state of a trigger and is the base class
+///        for trigger.
+class TriggerState
+{
+  public:
+    static constexpr uint64_t INVALID_TRIGGER_ID = std::numeric_limits<uint64_t>::max();
+    template <typename T>
+    using Callback = void (*)(T* const);
+
+    /// @brief constructs an empty TriggerState
+    TriggerState() = default;
+    virtual ~TriggerState() = default;
+
+    /// @brief constructs a TriggerState object
+    /// @param[in] origin pointer to the origin of the TriggerState
```

Comment:
Regarding documentation: it is e.g. not immediately clear what an origin of a TiggerState is supposed to be.

Reply:
Aligned the comments and the variable names with the ReadMe.md glossary